### DISCUSSION TITLE
GRID_STAT_CLIMO_MEAN_FILE_NAME Not Working as Expected in METplus 6

### DISCUSSION THREAD

**Discussion by lindsayblank** [dtcenter/METplus#2832] [Configuration] [answered]
Hello again, METplus folks,

I ran into an issue when upgrading workflows that calculate anomaly data from METplus5/MET 11 to METplus 6/MET 12. I set my climatology file for both forecast and obs using the GRID_STAT_CLIMO_MEAN_FILE_NAME wrapper, per the METplus 6-rc1 version of the [User's Guide](https://metplus.readthedocs.io/en/v6.0.0-rc1/Users_Guide/wrappers.html#gridstat). METplus 6 ran to completion with no errors, but my output was a file with only headers and no data. There were no error messages in my log file or a warning message stating that the field was depricated. 

When I set the climatology file separately for the forecast and ob using GRID_STAT_FCST_CLIMO_MEAN_FILE_NAME and GRID_STAT_OBS_CLIMO_MEAN_FILE_NAME respectively, METplus 6 again ran to completion with no errors, but this time, there was  my expected output. 

I wanted to make you aware of this behavior, and that, if it is correct, the change is not documented. Please let me know if you need any more information from me!

**Comment by georgemccabe** [comment]
Hi @lindsayblank,

That is not the behavior I would expect, so I'd like to look into this in more detail to determine if there is a bug. I would expect the old config setting to produce the same results.

Could you rerun both versions of METplus with `LOG_LEVEL=DEBUG` and compare the values that are set for `METPLUS_CLIMO_MEAN_DICT`? I would expect the same value to be set for both versions. You could provide those logs files as well if the issue is not immediately obvious. Thanks!

**Comment by lindsayblank** [comment]
Hi George,

Here are the results, as requested:
METplus 5.1/MET 11:
- Wrapper used: GRID_STAT_CLIMO_MEAN_FILE_NAME
- METPLUS_CLIMO_MEAN_DICT = climo_mean = {file_name = ["/scratch2/BMC/amb-verif/global_g2g_verif/climo/era5/mean_1109"];}

METplus 6/MET 12:
- Wrapper used: GRID_STAT_CLIMO_MEAN_FILE_NAME
- METPLUS_CLIMO_MEAN_DICT = climo_mean = {file_name = ["/scratch2/BMC/amb-verif/global_g2g_verif/climo/era5/mean_1109"]; regrid = {method = BILIN; width      = 2;} time_interp_method = NEAREST;match_month = TRUE;} --> this produces the .stat file with only headers, no output, and no error messages.

- Wrappers used: GRID_STAT_FCST_CLIMO_MEAN_FILE_NAME and GRID_STAT_OBS_CLIMO_MEAN_FILE_NAME
- METPLUS_CLIMO_MEAN_DICT=climo_mean = {regrid = {method = BILIN;width = 2;}time_interp_method = NEAREST;match_month = TRUE;}
- METPLUS_FCST_CLIMO_MEAN_DICT=climo_mean      = {file_name = ["/scratch2/BMC/amb-verif/global_g2g_verif/climo/era5/mean_1109"];}
-  METPLUS_OBS_CLIMO_MEAN_DICT=climo_mean      = {file_name = ["/scratch2/BMC/amb-verif/global_g2g_verif/climo/era5/mean_1109"];}

I don't see an obvious reason for why these are different, but I am running into some issues with scping the log files and am waiting for help. I will supply them when I can!

**Reply by JohnHalleyGotway** [reply]
@lindsayblank based on the paths, it looks like you're running on hera. Can you please point us to your config/run/log directories so we can go investigate further?

**Reply by lindsayblank** [reply]
Hi John, 

Sure thing. I apologize for not including the machine I was running on in my original message! 

Config directory: /home/role.amb-verif/GSL_verif-global_METplus6/config/metplus. The relevant config files are:
METplus 5/MET 11: global_g2g_gfs_consolidated_anom.conf
METplus 6/MET 12: global_g2g_gfs_consolidated_anom_METplus6.conf

Run directory: /scratch2/BMC/amb-verif/global_g2g_verif/g2g_reprocess/GFS/realtime/METplus6

Log directory: /scratch2/BMC/amb-verif/global_g2g_verif/g2g_reprocess/GFS/realtime/METplus6/logs
The logs of interest are in the logs_for_george directory. 

Please let me know if you need any other paths, like to the XML.

**Reply by JohnHalleyGotway** [reply, marked as answer]
@lindsayblank, thanks for sending along this location information. George and I looked through the details together and notice the following about the `logs_for_george`:

1. `metplus.log.20241216163747.metplus5_met11`
  - We note that you're using a custom Grid-Stat config file (`/home/role.amb-verif/GSL_verif-global_METplus6/config/met/GridStatConfig_GFS_anom`), and NOT the wrapped one for METplus.
  - This is working well because of this line that copies the list of fields to be used from the `fcst` dictionary into the `climo_mean` dictionary:
```
climo_mean = fcst;
```
2. `metplus.log.20241216170851.metplus6_met12_nooutput`
 - You're really doing 2 things here: switching METplus versions from 5 to 6 AND switching from using a custom MET config file to using the wrapped METplus one. The difference in behavior is related to both changes. Please try adding [this setting](https://metplus.readthedocs.io/en/main_v6.0/Users_Guide/glossary.html#term-GRID_STAT_CLIMO_MEAN_USE_FCST) to do the same thing via METplus: 
```
GRID_STAT_CLIMO_MEAN_USE_FCST = TRUE
```
3. `metplus.log.20241216171935.metplus6_met12_output`
  - The reason why this one DOES work is the following:
    - You set `GRID_STAT_FCST_CLIMO_MEAN_FILE_NAME` and `GRID_STAT_OBS_CLIMO_MEAN_FILE_NAME` explicitly.
    - That defines `climo_mean` information INSIDE the `fcst` and `obs` dictionaries in the MET config file.
    - You did NOT explicitly specify settings for [GRID_STAT_FCST_CLIMO_MEAN_FIELD](https://metplus.readthedocs.io/en/main_v6.0/Users_Guide/glossary.html#term-GRID_STAT_FCST_CLIMO_MEAN_FIELD) or [GRID_STAT_OBS_CLIMO_MEAN_FIELD](https://metplus.readthedocs.io/en/main_v6.0/Users_Guide/glossary.html#term-GRID_STAT_OBS_CLIMO_MEAN_FIELD).
    - Since Grid-Stat doesn't find settings in `fcst.climo_mean.field` it automatically uses the ones in `fcst.field` instead. And the same for the `obs` settings.

Please retry (2) above, using the `GRID_STAT_CLIMO_MEAN_USE_FCST = TRUE` and let us know if that yields the behavior you expect.

**Reply by lindsayblank** [reply]
Thank you for laying out the differences and explaining what is going on behind the scenes. I added `GRID_STAT_CLIMO_MEAN_USE_FCST = TRUE` to my METplus config file with the `GRID_STAT_CLIMO_MEAN_FILE_NAME` wrapper and I got the expected results using METplus 6/MET 12. Thank you again for your help with this!

**Reply by georgemccabe** [reply]
That's great to hear! I'm going to lock this discussion to prevent future posts, but please feel free to create a new discussion if you run into any other issues.
